app/models/concerns/canonical_email.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module CanonicalEmail
+  extend ActiveSupport::Concern
+
+  included do
+    normalizes :email, with: ->(value) { canonicalize(value) }
+  end
+
+  class_methods do
+    def canonicalize(email)
```

Comment:
Not sure I have a better idea, but I am a bit worried by this very generic name. Not a problem right now, but imagine this concern being used in multiple classes in the future.

In fact I think you had the same problem already. `::canonical_username` below has a more descriptive and unique name. Maybe this here should have that as well?

Reply:
Yeah - I see the point, and it's probably low risk given the single usage at this time. I did contemplate at various points in this branch history, things like:

- Just fold it all into the model, in which case even with `canonicalize` it's less likely to be accidentally used elsewhere
- Drop concern and move the "canonicalize email string" stuff to a utility class with something like `CanonicalEmail.new(email).to_s` or `CanonicalEmail.canonicalize(email)` or whatever called from the normalizer here

For now I've done the suggested `canonicalize_email` rename